Gem/Code/Source/Components/Multiplayer/GemComponent.cpp

```diff
@@ -89,9 +89,17 @@ namespace MultiplayerSample
 #if AZ_TRAIT_SERVER
     void GemComponentController::HandleRPC_CollectedByPlayer([[maybe_unused]] AzNetworking::IConnection* invokingConnection)
     {
-        // Hide the gem by moving it far away from the players' interest area.
-        // This removes the gem from the clients' view. See @sv_ClientAwarenessRadius.
-        GetParent().GetEntity()->GetTransform()->SetWorldTranslation(AZ::Vector3::CreateAxisZ(-1000.f));
+        if (m_controller)
+        {
+            m_controller->RemoveGem(m_gemTicketId);
+            m_controller = nullptr;
+        }
+    }
+
+    void GemComponentController::SetGemSpawnerController(GemSpawnerComponentController* controller, AzFramework::EntitySpawnTicket::Id gemTicketId)
+    {
+        m_controller = controller;
+        m_gemTicketId = gemTicketId;
```

Comment:
You shouldn't need to store the gem ticket id, you can just use GetEntity()->GetEntitySpawnTicketId() i think.

Reply:
Oh, cool!

Reply:
Done.